### PR TITLE
secondary compartment/context assignment

### PR DESCRIPTION
**Pull request by a-w-beck**

### CODE REVIEW

File: esupy/context_secondary.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        log.error('Please pass one or more valid *cmpts string codes: {urb, rh}')
+        return df
+    elif 'urb' in cmpts and not has_geo_pkgs:
+        log.info('Geospatial dependencies of esupy.context_secondary missing; '
+                  'unable to assign urban/rural compartment.\n See esupy README.md.')
+        return df
+    if 'urb' in cmpts:
+        df = urb_intersect(df, year)
+    if 'rh' in cmpts:
+        df = classify_height(df)
```
Comment (bl-young) [inline esupy/context_secondary.py:173]: would it make sense to move `classify_height()` up prior to checking `has_geo_pkgs`? That way stack height can still be assigned even if geospatial packages are not available.

Comment (a-w-beck) [reply esupy/context_secondary.py:173]: These separate if-statements can already handle assigning urb, rh, or both based on the contents of the `*cmpts` tuple. If `has_geo_pkgs` = False and `"rh' in cmpts` is True, then `classify_height()` is still called.

Comment (bl-young) [reply esupy/context_secondary.py:173]: But in the case of NEI where both `urb` and `rh` are passed, if the packages are not installed line 169 will return the unaltered dataframe

Comment (a-w-beck) [reply esupy/context_secondary.py:173]: Ah yes okay, I'll go ahead and drop 169 and add `and has_geo_pkgs` to 170

**Pull request merged**